IconThemePlugin.js

```diff
@@ -0,0 +1,37 @@
+class IconThemePlugin {
+  // Define `apply` as its prototype method which is supplied with compiler as its argument
+  apply(compiler) {
+    const pluginName = IconThemePlugin.name;
+    const { webpack } = compiler;
+    const { RawSource } = webpack.sources;
+
+    // Specify the event hook to attach to
+    compiler.hooks.emit.tap(pluginName, (compilation) => {
+      const icons = compilation.getAssets().map(
+        (asset) => asset.name.split('/').pop()
+      );
+
+      // Create an object with the property name as the asset name
+      // and the value as the file extension
+      const iconTypes = icons.reduce(
+        (acc, icon) => {
+          const [name, ext] = icon.split('.');
+          acc[name] = ext;
+          return acc;
+        },
+        {}
+      );
+
+      // Read the current metadata file
```

Comment:
I don't think the comments here are needed. It's pretty self explanatory, especially if the above reduce is replaced.

Reply:
No problem, will remove 👍🏼